core/domain/feedback_services.py

```diff
@@ -406,16 +437,67 @@ def clear_feedback_message_references(user_id, exploration_id, thread_id):
         model.put()
 
 
+def get_all_recipient_list(exploration_id, thread_id, author_id):
+    """Returns dict containing batch recipients and other recipients
+    for sending feedback message email.
+    """
+    exploration_rights = rights_manager.get_exploration_rights(exploration_id)
+
+    owner_list = set(exploration_rights.owner_ids)
+    participant_list = get_all_thread_participants(exploration_id, thread_id)
+    sender = set([author_id])
+
+    batch_recipients = owner_list - sender
+    other_recipients = participant_list - batch_recipients - sender
+
+    return {
+        'batch_recipients': batch_recipients,
```

Comment:
Actually, given how you're using this, you might also consider returning a 2-tuple. Then you have:

    batch_recipient_ids, other_recipient_ids = get_all_recipient_list(
        exploration_id, thread_id, author_id)

at the callsite, which looks quite clean.